src/data/mystery-encounters/encounters/lost-at-sea-encounter.ts

```diff
@@ -0,0 +1,144 @@
+import { Type } from "#app/data/type.js";
+import { Species } from "#app/enums/species.js";
+import { PlayerPokemon } from "#app/field/pokemon.js";
+import { MysteryEncounterType } from "#enums/mystery-encounter-type";
+import BattleScene from "../../../battle-scene";
+import MysteryEncounter, {
+  MysteryEncounterBuilder,
+  MysteryEncounterTier,
+} from "../mystery-encounter";
+import { MysteryEncounterOptionBuilder } from "../mystery-encounter-option";
+import {
+  applyDamageToPokemon,
+  leaveEncounterWithoutBattle,
+  setEncounterExp,
+} from "../mystery-encounter-utils";
+
+/**
+ * Damage percentage taken when wandering aimlessly.
+ * Can be a number between `0` - `100`.
+ * The higher the more damage taken (100% = instant KO).
+ */
+const DAMAGE_PERCENTAGE: number = 30; // 0 - 100
+
+let waterPkm: PlayerPokemon;
+let flyingPkm: PlayerPokemon;
+
+/**
+ * Lost at sea encounter.
+ * @see {@link https://github.com/AsdarDevelops/PokeRogue-Events/issues/9 | GitHub Issue #9}
+ * @see For biome requirements check [mysteryEncountersByBiome](../mystery-encounters.ts)
+ */
+export const LostAtSeaEncounter: MysteryEncounter =
+  MysteryEncounterBuilder.withEncounterType(MysteryEncounterType.LOST_AT_SEA)
+    .withEncounterTier(MysteryEncounterTier.COMMON)
+    .withIntroSpriteConfigs([
+      {
+        fileRoot: "pokemon",
+        spriteKey: `${Species.GYARADOS}`,
+        hasShadow: false,
+        scale: 4,
+        y: 100,
+        x: 130,
+        tint: 0.75,
+        alpha: 0.25,
+      },
+    ])
+    .withSceneWaveRangeRequirement(11, 179)
+    .withOnInit((scene: BattleScene) => {
+      const allowedPokemon = scene
+        .getParty()
+        .filter((p) => p.isAllowedInBattle());
+      const { mysteryEncounter } = scene.currentBattle;
+
+      mysteryEncounter.setDialogueToken(
+        "damagePercentage",
+        String(DAMAGE_PERCENTAGE)
+      );
+
+      // check for water pokemon
+      waterPkm = findPokemonByType(allowedPokemon, Type.WATER);
+      mysteryEncounter.setDialogueToken("waterPkm", waterPkm?.name ?? "<NONE>");
+
+      // check for flying pokemon
+      flyingPkm = findPokemonByType(allowedPokemon, Type.FLYING);
+      mysteryEncounter.setDialogueToken(
+        "flyingPkm",
+        flyingPkm?.name ?? "<NONE>"
+      );
+
+      return true;
+    })
+    /**
+     * Option 1: Use a (non fainted) water pokemon to guide you back.
+     * Receives EXP similar to defeating a Lapras
+     */
+    .withOption(
+      new MysteryEncounterOptionBuilder()
+        .withPokemonTypeRequirement(Type.WATER, true, 1)
```

Comment:
Remember it should be "can learn Surf" in the final version, not "be water type". Same for flying type, with Fly move

Reply:
meh...

Reply:
done